grails-app/controllers/UserProfileController.groovy

```diff
@@ -0,0 +1,100 @@
+import org.transmart.searchapp.AuthUser
+import org.transmartproject.db.log.AccessLogService
+
+class UserProfileController {
+
+	def springSecurityService
+	def userProfileService
+	def auth0Service
+	def userService
+	def auth0Config
+	AccessLogService accessLogService
+
+	// non-recoverable error
+	String severeMessage = "Unable to update user information. Contact administrator."
+
+	def index() {
+		try{
+			def userDetails
+			def token
+
+			token = auth0Service.jwtToken()
+			userDetails = userService.currentUserInfo()
+
+            model:[
+				user: userDetails,
+				token: token?: "Unable to retrieve token.",
+				instanceType: auth0Config.instanceType,
+				instanceName: auth0Config.instanceName,
+				level: userDetails.level
+			]
+
+		}
+		catch (Exception e) {
+			log.error("Caught error in UserProfile plugin.", e);
```

Comment:
To make the Grails 3 upgrade simpler, use Slf4j instead of the `log` variable injected by Grails. Annotate the class with `@Slf4j('logger')` (the name is overridden to 'logger' to avoid accidentally calling `log`) and use string (not GString) messages with `{}` placeholders, e.g. `logger.error 'Caught error {} in UserProfile plugin', e.message, e`